robot/packages/cloud_package_manager.go

```diff
@@ -238,58 +195,110 @@ func (m *cloudManager) Cleanup(ctx context.Context) error {
 
 	var allErrors error
 
-	// packageDir will contain either symlinks to the packages or the .data directory.
-	files, err := os.ReadDir(m.packagesDir)
+	expectedPackageDirectories := map[string]bool{}
+	for _, pkg := range m.managedPackages {
+		expectedPackageDirectories[pkg.thePackage.LocalDataDirectory()] = true
+	}
+
+	topLevelFiles, err := os.ReadDir(m.packagesDir)
 	if err != nil {
 		return err
 	}
-
-	// keep track of known packages by their hashed name from the package id and version.
-	knownPackages := make(map[string]bool)
-
-	// first remove all symlinks to the packages themself.
-	for _, f := range files {
-		if f.Type()&os.ModeSymlink == os.ModeSymlink {
-			// if managed skip removing package
-			if p, ok := m.managedPackages[PackageName(f.Name())]; ok {
-				knownPackages[p.thePackage.SanitizeName()] = true
-				continue
+	// A packageTypeDir is a directory that contains all of the packages for the specified type. ex: packages/ml_models
+	for _, packageTypeDir := range topLevelFiles {
+		if packageTypeDir.Type()&os.ModeDir != os.ModeDir {
+			continue
+		}
```

Comment:
I assume you are adding this in case a file is created by a user? Wondering if we really need to be this failsafe for unexpected cases like this. (instead assume that the directory is only being touched by the RDK).

Reply:
Without this check, the behavior of the function is more complicated (ex: I call `os.ReadDir` on something that might not be a directory so now I have to think about what the errors would look like) (or what happens if a user adds a symlink out of /home/.viam and then this deletes random other files). This check makes the code easier to read, while also being safer (imo)

Reply:
logic in this function has changed a bit. resolving